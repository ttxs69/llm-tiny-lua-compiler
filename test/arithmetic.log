Advanced to token TOKEN_PRINT 'print'
Advanced to token TOKEN_LPAREN '('
Advanced to token TOKEN_NUMBER '1'
Advanced to token TOKEN_PLUS '+'
Advanced to token TOKEN_NUMBER '2'
Advanced to token TOKEN_MUL '*'
Advanced to token TOKEN_NUMBER '3'
Advanced to token TOKEN_RPAREN ')'
Advanced to token TOKEN_PRINT 'print'
Advanced to token TOKEN_LPAREN '('
Advanced to token TOKEN_LPAREN '('
Advanced to token TOKEN_NUMBER '1'
Advanced to token TOKEN_PLUS '+'
Advanced to token TOKEN_NUMBER '2'
Advanced to token TOKEN_RPAREN ')'
Advanced to token TOKEN_MUL '*'
Advanced to token TOKEN_NUMBER '3'
Advanced to token TOKEN_RPAREN ')'
Advanced to token TOKEN_PRINT 'print'
Advanced to token TOKEN_LPAREN '('
Advanced to token TOKEN_NUMBER '10'
Advanced to token TOKEN_MINUS '-'
Advanced to token TOKEN_NUMBER '4'
Advanced to token TOKEN_DIV '/'
Advanced to token TOKEN_NUMBER '2'
Advanced to token TOKEN_MUL '*'
Advanced to token TOKEN_NUMBER '3'
Advanced to token TOKEN_RPAREN ')'
Advanced to token TOKEN_PRINT 'print'
Advanced to token TOKEN_LPAREN '('
Advanced to token TOKEN_NUMBER '5'
Advanced to token TOKEN_DIV '/'
Advanced to token TOKEN_MINUS '-'
Advanced to token TOKEN_NUMBER '2'
Advanced to token TOKEN_PLUS '+'
Advanced to token TOKEN_NUMBER '2'
Advanced to token TOKEN_RPAREN ')'
Advanced to token TOKEN_PRINT 'print'
Advanced to token TOKEN_LPAREN '('
Advanced to token TOKEN_NUMBER '2.5'
Advanced to token TOKEN_MUL '*'
Advanced to token TOKEN_NUMBER '2'
Advanced to token TOKEN_RPAREN ')'
Advanced to token TOKEN_EOF ''
