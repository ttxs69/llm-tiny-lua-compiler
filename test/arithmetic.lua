print(1 + 2 * 3)
print((1 + 2) * 3)
print(10 - 4 / 2 * 3)
print(5 / -2 + 2)
print(2.5 * 2)
