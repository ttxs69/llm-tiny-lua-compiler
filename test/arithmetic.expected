7.000000
9.000000
8.000000
-3.000000
5.000000
